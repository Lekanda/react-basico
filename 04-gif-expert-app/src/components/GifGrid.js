import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs =async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morti&limit=10&api_key=Fy5o4SLF3Q2ij74a1bG1N3NoL20R8Cv8';

        const resp = await fetch(url);
        const {data} = await resp.json();
        
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
