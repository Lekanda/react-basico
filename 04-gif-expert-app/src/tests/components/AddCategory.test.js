import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from "../../components/AddCategory";

describe('Probar Componente AddCategory', () => {

    const setCategories = () => {};

    test('Debe de mostrarse correctamnete en el snapshot', () => {
        const wrapper = shallow(<AddCategory setCategories={ setCategories } /> );
        expect(wrapper).toMatchSnapshot();
    })
});