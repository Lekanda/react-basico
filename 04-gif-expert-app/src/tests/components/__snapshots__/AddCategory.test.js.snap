// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar Componente AddCategory Debe de mostrarse correctamnete en el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
